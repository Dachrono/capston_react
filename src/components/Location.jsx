import '../style/Location.css';
import { useNavigate } from 'react-router-dom';

const Location = () => {
  const nav = useNavigate();

  const selectValue = (event) => {
    nav(event.target.value);
  };

  return (
    <div className="titleContainer">
      <h1>Metrics | Wheather</h1>
      <select className="filter" defaultValue="0" onChange={selectValue}>
        <option value="0" disabled>Choose your region</option>
        <option value="/Canada">Canada</option>
        <option value="/United States">United States</option>
        <option value="/Mexico">Mexico</option>
        <option value="/Brazil">Brazil</option>
        <option value="/Argentina">Argentina</option>
        <option value="/Chile">Chile</option>
      </select>
    </div>
  );
};

export default Location;
