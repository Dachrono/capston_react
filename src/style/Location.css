.titleContainer {
  margin: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 40vw;
  background-color: #fd5092;
  color: #fff;
  background-image: url('../assets/world.png');
  background-size: 100% 100%;
}

select {
  background-color: transparent;
  font-size: 20px;
  color: #313131;
}
