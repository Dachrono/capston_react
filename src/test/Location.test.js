import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Location from '../components/Location';
import '@testing-library/jest-dom/extend-expect';

describe('Test for the location component', () => {
  test('Check if the location component renders correctly', () => {
    const { container } = render(<MemoryRouter><Location /></MemoryRouter>);
    expect(container).toMatchSnapshot();
  });

  test('Test the content of select', () => {
    const { container } = render(<MemoryRouter><Location /></MemoryRouter>);
    expect(container.querySelector('select')).toHaveTextContent('Canada');
  });
});
