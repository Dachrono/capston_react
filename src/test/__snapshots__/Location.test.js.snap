// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the location component Check if the location component renders correctly 1`] = `
<div>
  <div
    class="titleContainer"
  >
    <h1>
      Metrics | Wheather
    </h1>
    <select
      class="filter"
    >
      <option
        disabled=""
        selected=""
        value="0"
      >
        Choose your region
      </option>
      <option
        value="/Canada"
      >
        Canada
      </option>
      <option
        value="/United States"
      >
        United States
      </option>
      <option
        value="/Mexico"
      >
        Mexico
      </option>
      <option
        value="/Brazil"
      >
        Brazil
      </option>
      <option
        value="/Argentina"
      >
        Argentina
      </option>
      <option
        value="/Chile"
      >
        Chile
      </option>
    </select>
  </div>
</div>
`;
